src/main/java/org/broadinstitute/hellbender/tools/validation/CompareBaseQualities.java

```diff
@@ -0,0 +1,85 @@
+package org.broadinstitute.hellbender.tools.validation;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.SecondaryOrSupplementarySkippingIterator;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.hellbender.cmdline.*;
+import org.broadinstitute.hellbender.cmdline.programgroups.ReadProgramGroup;
+import org.broadinstitute.hellbender.engine.ProgressMeter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.util.List;
+
+@CommandLineProgramProperties(
+        summary = "USAGE: CompareBaseQualities <SAMFile1> <SAMFile2>\n" +
+                "Compares the base qualities of two input SAM/BAM/CRAM files. The files must be sorted exactly the same name.",
+        oneLineSummary = "Compares base qualities of two input SAM/BAM/CRAM files",
+        programGroup = ReadProgramGroup.class
+)
+public final class CompareBaseQualities extends PicardCommandLineProgram {
+    @PositionalArguments(minElements = 2, maxElements = 2)
+    public List<File> samFiles;
+
+    @Argument(doc="Whether to allow  differences in read order", shortName = "allow_different_read_order", fullName = "allow_different_read_order")
+    private boolean allowDifferentReadOrder= false;
```

Comment:
Can you add a comment on what the expected behavior is when `allowDifferentReadOrder` is `true` (better yet, also add under what circumstances one would want that -- instead of just sorting both files before running).